File: src/SIStorage.Service.Contract/Requests/RandomPackageParameters.cs
namespace SIStorage.Service.Contract.Requests;

/// <summary>
/// Defines random package parameters.
/// </summary>
public sealed record RandomPackageParameters
{
    /// <summary>
    /// Package tags.
    /// </summary>
    public int[]? TagIds { get; set; }

    /// <summary>
    /// Package difficulty.
    /// </summary>
    public short? Difficulty { get; set; }

    /// <summary>
    /// Package restrictions.
    /// Restrictions with the same name are joined with OR logic.
    /// Restrictions with different names are joined with AND logic.
    /// </summary>
    public int[]? RestrictionIds { get; set; }

    /// <summary>
    /// Package language.
    /// </summary>
    public int? LanguageId { get; set; }

    /// <summary>
    /// Round count.
    /// </summary>
    public int RoundCount { get; set; } = 3;

    /// <summary>
    /// Table round theme count.
    /// </summary>
    public int TableThemeCount { get; set; } = 6;

    /// <summary>
    /// Theme list round theme count.
    /// </summary>
    public int ThemeListThemeCount { get; set; } = 7;

    /// <summary>
    /// Base question price.
    /// </summary>
    public int BaseQuestionPrice { get; set; } = 100;
}

File: web/src/models/RandomPackageParameters.ts
/** Defines random package parameters. */
export default interface RandomPackageParameters {
	/** Package tags. */
	tagIds?: number[];

	/** Package difficulty. */
	difficulty?: number;

	/**
	 * Package restrictions.
	 * Restrictions with the same name are joined with OR logic.
	 * Restrictions with different names are joined with AND logic.
	 */
	restrictionIds?: number[];

	/** Package language. */
	languageId?: number;

	/** Round count. */
	roundCount?: number;

	/** Table round theme count. */
	tableThemeCount?: number;

	/** Theme list round theme count. */
	themeListThemeCount?: number;

	/** Base question price. */
	baseQuestionPrice?: number;
}